api/solarperformanceinsight_api/compute.py

```diff
@@ -529,9 +539,36 @@ def _get_temp(
     return tuple(out)
 
 
+def _get_missing_leap_days(dfs: List[pd.DataFrame]) -> Set[dt.datetime]:
+    if len(dfs) == 0:
+        return set()
+
+    per_df = []
+    for df in dfs:
+        thisset = set()
+        if not df.index.is_leap_year.any():  # type: ignore
+            return set()
+        leap_df = df.loc[df.index.is_leap_year]  # type: ignore
+
+        for _, grp in leap_df.groupby(leap_df.index.year):  # type: ignore
+            feb29 = grp.index.is_leap_year & (grp.index.dayofyear == 60)  # type: ignore
+            if not feb29.any():
+                continue
+            nans = pd.isnull(grp.loc[feb29])
+            # if not all data is nan on feb29, don't drop
+            if not nans.all().all():
+                continue
+            thisset |= set(grp.index[feb29].to_list())
```

Comment:
```suggestion
            # all data is nan on feb29, so user did not include date in
            # upload (likely) or just specified nan values (less likely)
            if nans.all().all():
                thisset |= set(grp.index[feb29].to_list())
```

Reply:
This is based on my understanding that data uploaded by a user is reindexed to `pd.date_range(start, end, freq)`, so nans are inserted into the data.